core/controllers/editor.py

```diff
@@ -889,43 +888,35 @@ def post(self, exploration_id):
         if not raw:
             raise self.InvalidInputException('No image supplied')
 
-        file_format = imghdr.what(None, h=raw)
```

Comment:
I think you're right.

Here's what I've done:

- I put back the imghrd checking, but only to verify that the data given looks like image data. I do not rely on this library to tell me which specific format / extension to associated to the data. Sounds good? So we're still catching non-image data being passed.
- I put back the removed test, which passes non-image data and verifies that the proper error gets triggered.